GeometryCore/src/Polygon.cpp

```diff
@@ -27,8 +27,20 @@ namespace geom_utils
 
     // Return true if polygon orientation is counter-clockwise (i.e. area is positive)
     bool Polygon::orientation() const
-    { 
-        return {}; 
+    {
+        //if (this->points.size() < 3) return false;
+        float res = 0.0;
+
+        #pragma omp parallel for
+        for (int i = 0; i < points.size(); i++)
+        {
+            FPoint2D v1 = points[i];
+            FPoint2D v2 = points[(i + 1) % points.size()];
+            #pragma omp atomic
+            res += (v2.x - v1.x) * (v2.y + v1.y);
+        }
+
+        return res < 0 ? true : false;
```

Comment:
`return res < 0 ? true : false;` is verbose. `return res < 0;` will have the same effect.

Reply:
oh, it is really better, i will implement it